mailbox/api/src/test/java/org/apache/james/mailbox/MailboxManagerStressContract.java

```diff
@@ -41,45 +41,37 @@
 import org.apache.james.mailbox.model.MailboxPath;
 import org.apache.james.mime4j.dom.Message;
 import org.apache.james.util.concurrent.NamedThreadFactory;
-import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
 import com.google.common.collect.ImmutableSet;
 
-public abstract class MailboxManagerStressTest<T extends MailboxManager> {
+public interface MailboxManagerStressContract<T extends MailboxManager> {
 
-    private static final int APPEND_OPERATIONS = 200;
+    int APPEND_OPERATIONS = 200;
 
-    private T mailboxManager;
+    T getManager();
 
-    protected abstract T provideManager() throws MailboxException;
-
-    protected abstract EventBus retrieveEventBus(T mailboxManager);
-
-    @BeforeEach
-    void setUp() throws Exception {
-        this.mailboxManager = provideManager();
-    }
+    EventBus retrieveEventBus(T mailboxManager);
```

Comment:
IMO we should let the underlying testing class give it's EventBus through a `EventBus getEventBus()` method.